lib/perl/Genome/Model/Tools/Manta/Base.pm

```diff
@@ -0,0 +1,246 @@
+    my $type = defined $meta->data_type ? $meta->data_type : 'String';
+    my $tool_arg_name = $meta->tool_arg_name;
+
+    return unless defined $self->$ur_name;
+    if ($type eq 'Boolean' && $self->$ur_name eq '0') {
+        return;
+    }
+
+    # This works for things that are is_many or not.
+    return map {_format_tool_arg($type, $tool_arg_name, $_)} $self->$ur_name;
+}
+
+sub _tool_input_files_from_meta {
+    my ($self, $meta) = @_;
+
+    my $ur_name = $meta->property_name;
+
+    return unless defined $self->$ur_name;
+
+    # This works for things that are is_many or not.
+    return map {$_} $self->$ur_name;
+}
+
+# command line args that come after the executable
+#
+# The default implementation just translates all the inputs that have
+# speedseq_param_name defined. Boolean values get translated to true/false,
+# and is many attributes get repeated, e.g.,
+#
+#  create_md5_file => {..., is => 'Boolean'}
+#       when set to something that evalutes to false
+#       yields
+#  qw(CREATE_MD5_FILE=false)
+#
+# and
+#
+#  program_name => {..., is_many => 1}
+#       when set to ['a', 'b', 'c']
+#       yields
+#  qw(PROGRAM_NAME=a PROGRAM_NAME=b PROGRAM_NAME=c)
+#
+
+sub _tool_param_args {
+    my $self = shift;
+
+    my @metas = $self->_tool_param_metas;
+    my @args = map {$self->_tool_args_from_meta($_)} @metas;
+
+    return @args;
+}
+
+sub _tool_input_args {
+    my $self = shift;
+
+    my @metas = $self->_tool_input_metas;
+    my @args = map {$self->_tool_args_from_meta($_)} @metas;
+
+    return @args;
+}
+
+# this is your chance to make a loud sound and die if a caller asks for
+# something ridiculous. params can also be edited here if that is somehow
+# appropriate.
+sub _validate_params {
+    # example:
+    # my $self = shift;
+    # $self->enforce_minimum_version('1.85');
+    # die unless -s $self->input_file;
+    # $self->be_noisy(0) unless $self->log_file;
+
+    #TODO: enforce one of bams, normal_bam or tumor_bam inputs are required
+}
+
+# want to pass extra stuff to Genome::Sys->shellcmd? return a hash (not ref)
+# of those params here.
+sub _shellcmd_extra_params {
+    my $self = shift;
+    my @output_files = $self->_resolve_output_files;
+    my @input_files = $self->_resolve_input_files;
+    return (
+        skip_if_output_is_present => 0,
+        keep_dbh_connection_open => 0,
+        set_pipefail => 0,
+        input_files => \@input_files,
+        output_files => \@output_files,
+    );
+}
+
+sub build_cmdline_list {
+    my $self = shift;
+
+    return (
+        $self->tool_path,
+        $self->_tool_param_args,
+        $self->_tool_input_args,
+    );
+}
+
+sub build_cmdline_string {
+    my $self = shift;
+
+    my $cmd = join(' ', $self->build_cmdline_list);
+    return $cmd;
+}
+
+# override with care.
+sub execute {
+    my $self = shift;
+
+    $self->_validate_params;
+
+    my %params = (
+        cmd => $self->build_cmdline_string,
```

Comment:
Looks like the list implementation in this tool is broken.  Specifically `_format_tool_arg` makes a string instead of building up a list of arguments.